lib/optimizely/helpers/event_tag_utils.rb

```diff
@@ -21,37 +21,86 @@ module Helpers
     module EventTagUtils
       module_function
 
+      REVENUE_EVENT_METRIC_NAME = 'revenue';
+      VALUE_EVENT_METRIC_NAME = 'value';
+
       def get_revenue_value(event_tags)
         # Grab the revenue value from the event tags. "revenue" is a reserved keyword.
         #
-        # event_tags - Hash representing metadata associated with the event.
-        # Returns revenue value as an integer number
-        # Returns nil if revenue can't be retrieved from the event tags.
+        # event_tags - +Hash+ representing metadata associated with the event.
+        #
+        # Returns +Integer+ | +nil+ if revenue can't be retrieved from the event tags.
 
-        if event_tags.nil? or !Helpers::Validator.attributes_valid?(event_tags)
-          return nil
-        end
+        revenue_value = nil
+        
+        if ( event_tags and Helpers::Validator.attributes_valid?(event_tags) and event_tags.has_key?(REVENUE_EVENT_METRIC_NAME) )
+  
+          logger = SimpleLogger.new
+  
+          begin
+            raw_value = event_tags[REVENUE_EVENT_METRIC_NAME]
 
-        unless event_tags.has_key?('revenue')
-          return nil
-        end
+            if raw_value.is_a? Numeric
+              num_value = raw_value
+            else
+              num_value = raw_value.to_i
+            end
+
+            if (!num_value.is_a?(Float)) and num_value.to_s == raw_value.to_s
+              revenue_value = raw_value
+              logger.log(Logger::INFO, "Parsed revenue value #{raw_value} from event tags.")
+            else
+              logger.log(Logger::WARN, "Failed to parse revenue value #{raw_value} from event tags.")
+            end
 
-        logger = SimpleLogger.new
-        raw_value = event_tags['revenue']
+          rescue
+            logger.log(Logger::WARN, "Failed to parse revenue value #{raw_value} from event tags.")
+          end
 
-        unless raw_value.is_a? Numeric
-          logger.log(Logger::WARN, "Failed to parse revenue value #{raw_value} from event tags.")
-          return nil
         end
 
-        if raw_value.is_a? Float
-          logger.log(Logger::WARN, "Failed to parse revenue value #{raw_value} from event tags.")
-          return nil
+        revenue_value
+      end
+
+      def get_event_value(event_tags)
```

Comment:
this function should have logger, for now we are not passing any logger from any function, but we need to pass using function.

Reply:
@faisal-iqbal  please copy the same code & messages from c-sharp or python sdk.

Reply:
change the method name from GetNumericValue

Reply:
function was named as https://github.com/optimizely/node-sdk/blob/94f940cff1872ec1023fd9eb2ad4a3781d31d61d/lib/utils/event_tag_utils/index.js#L56

logger are not sent as parameter in ruby sdk see: get_revenue_value function for reference

Reply:
@faisal-iqbal They have directed on Csharp Pull Request to do how it has been done in the Python SDK : See here
https://github.com/optimizely/python-sdk/blob/master/optimizely/helpers/event_tag_utils.py#L40
Also, the logger needs to be added. 
And, the logger be passed from unit tests. As in here:
https://github.com/optimizely/python-sdk/blob/master/tests/helpers_tests/test_event_tag_utils.py